Apps/Src/Temperature.c

```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
 
 /**
  * @brief Gives fan speed based on Average temperature of pack and past error values
- * @param InputTemp - current temperature
+ * @param InputTemp - current temperature. Must not exceed 284 or -284 Celcius given that expected temp is 38 celcius.
  * @param DesiredTemp - desired temperature
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
-int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
+int8_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
     Error = DesiredTemp - InputTemp;
-    ErrorSum = ErrorSum + Error;
+	
+    //Only read error sum in range
+    ErrorSum += abs(Error) < TEMPERATURE_PID_I_ZONE ? Error : 0;
```

Comment:
why is this here, reading errors only in range of 3 C? Temperature_PID_I_ZONE is only 3000 which is 3 celsius. idk but feels low to me

Reply:
3 was a guess, i actually just increased it to 5. Regardless this number is very low because you don't want to use the "i-part" of the pid unless you are very close to the target. 

Error, instead of errorsum, is used by the "p-part" and is what is read and updated regardless of how close to the target we are.

Reply:
jk going back to three